### PR TITLE
feat(inverted_index.format): add writer

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Add format writer, accepts stream of values.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#2705

### CODE REVIEW

File: src/index/src/inverted_index/error.rs [hunk 1]
```diff
@@ -115,6 +136,20 @@ pub enum Error {
 
     #[snafu(display("index not found, name: {name}"))]
     IndexNotFound { name: String, location: Location },
+
+    #[snafu(display("Failed to invert value to FST"))]
```
Comment (waynexia) [inline src/index/src/inverted_index/error.rs:140]:
```suggestion
    #[snafu(display("Failed to insert value to FST"))]
```

**Review by waynexia — APPROVED** [review]
Others LGTM

File: src/index/src/inverted_index/format.rs [hunk 2]
```diff
@@ -27,10 +27,10 @@
 //!
 //! An inverted index comprises a collection of bitmaps, a null bitmap, and a finite state transducer (FST) indicating tag values' positions:
 //!
-//! `bitmap₀ bitmap₁ bitmap₂ ... bitmapₙ null_bitmap fst`
+//! `null_bitmap bitmap₀ bitmap₁ bitmap₂ ... bitmapₙ fst`
```
Comment (killme2008) [inline src/index/src/inverted_index/format.rs:30]: Looks like we move forward the null_bitmap before bitmaps, can you explain why?

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by evenyag — APPROVED** [review]

File: src/index/src/inverted_index/format.rs (hunk 2 above)
Comment (zhongzc) [reply src/index/src/inverted_index/format.rs:30]: 1. For Read, bitmap reading relies on offset and size. Whether `null_bitmap` comes first or not does not affect the format and will not destroy the implementation.
2. For Write, some implementations of index building may directly use `BTreeMap<Option<Bytes>, BitVec>>`. In this case, it is more natural to write `null_bitmap` first. (Of course, this is not the case in my implementation, but I'm happy to handle nulls first)

Comment (killme2008) [reply src/index/src/inverted_index/format.rs:30]: Thank you for the info.

**Pull request merged**